openquakeplatform/vulnerability/admin.py

```diff
@@ -0,0 +1,1163 @@
+admin.site.register(DamageToLossFunc, Hide2StaffAdmin)
+admin.site.register(FuncDistrVulnCont, Hide2StaffAdmin)
+admin.site.register(FuncDistrVulnDiscr, Hide2StaffAdmin)
+admin.site.register(EvaluationOfIM, Hide2StaffAdmin)
+admin.site.register(PredictorVar, Hide2StaffAdmin)
+admin.site.register(VulnerabilityFunc, Hide2StaffAdmin)
+admin.site.register(FuncDistrFragCont, Hide2StaffAdmin)
+admin.site.register(FuncDistrFragDiscr, Hide2StaffAdmin)
+admin.site.register(EngineeringDemandPar, Hide2StaffAdmin)
+admin.site.register(FragilityFunc, Hide2StaffAdmin)
+admin.site.register(GeneralInformation, Hide2StaffAdmin)
+
+
+from models import TA, TYPES_OF_ASSESSMENT, FDT
+
+
+class NestedModelAdminCrossChecked(NestedModelAdmin):
+    def is_cross_valid(self, form, formsets):
+        return True
+
+    # response_add and response_change methods use '_post' methods to redirect
+    def response_post_save_add(self, request, obj):
+        """
+        Figure out where to redirect after the 'Save' button has been pressed
+        when adding a new object.
+        """
+        if self.has_change_permission(request, None):
+            post_url = "/vulnerability/view/%d/" % obj.id
+            return HttpResponseRedirect(post_url)
+        else:
+            return super(NestedModelAdminCrossChecked, self).response_post_save_add(request, obj)
+
+    def response_post_save_change(self, request, obj):
+        """
+        Figure out where to redirect after the 'Save' button has been pressed
+        when editing an existing object.
+        """
+        if self.has_change_permission(request, None):
+            post_url = "/vulnerability/view/%d/" % obj.id
+            return HttpResponseRedirect(post_url)
+        else:
+            return super(NestedModelAdminCrossChecked, self).response_post_save_change(request, obj)
+
+    # delete_view method has no post_delete method associated then override
+    # it modifying the django.contrib.admin.options.ModelAdmin
+    @csrf_protect_m
+    @transaction.atomic
+    def delete_view(self, request, object_id, extra_context=None):
+        "The 'delete' admin view for this model."
+        opts = self.model._meta
+        app_label = opts.app_label
+
+        obj = self.get_object(request, unquote(object_id))
+        type_of_assessment = obj.type_of_assessment
+        if not self.has_delete_permission(request, obj):
+            raise PermissionDenied
+
+        if obj is None:
+            raise Http404(
+                _('%(name)s object with primary key %(key)r does not exist.') %
+                {'name': force_unicode(opts.verbose_name),
+                 'key': escape(object_id)})
+
+        using = router.db_for_write(self.model)
+
+        # Populate deleted_objects, a data structure of all related objects
+        # that will also be deleted.
+        (deleted_objects, model_count, perms_needed, protected) = get_deleted_objects(
+        #deleted_objects = get_deleted_objects(
```

Comment:
Please remove it.

Reply:
Fixed here: https://github.com/gem/oq-platform2/pull/19/commits/f440b714cebf6d76ac9232655e33cbfe5237d8a0